Problems compiling for Mac

Sorry but I don't do much command line compiling.

I have openssl installed using `brew` at `/usr/local/opt/openssl/`

```
$ make
g++  -c maketmd.cpp -o maketmd.o
maketmd.cpp:37:10: fatal error: 'openssl/sha.h' file not found
#include <openssl/sha.h>
         ^~~~~~~~~~~~~~~
1 error generated.
make: *** [maketmd.o] Error 1
```

I added `CFLAGS = "-I/usr/local/opt/openssl/include"` the following to the Makefile

```
$ make
g++ "-I/usr/local/opt/openssl/include" -c maketmd.cpp -o maketmd.o
maketmd.cpp:44:10: fatal error: 'byteswap.h' file not found
#include <byteswap.h>
         ^~~~~~~~~~~~
1 error generated.
make: *** [maketmd.o] Error 1
```

at this point I fear I'm heading down a rabbit hole, as `glibc` has not been ported to macOS

any help appreciated!

--- a/.gitignore
+++ b/.gitignore
@@ -30,3 +30,6 @@
 *.exe
 *.out
 *.app
+
+# MacOS annoying file
+.DS_Store

--- a/Makefile
+++ b/Makefile
@@ -16,8 +16,8 @@ endif
 all: $(TARGET)
 
 version.h: Makefile
-	@echo -e "#define TMD_CREATOR_VER  \""$(VERSION_VER)"\"" > version.h
-	@echo -e "#define TMD_CREATOR_DATE \""$(VERSION_DATE)"\"" >> version.h
+	@printf "#define TMD_CREATOR_VER  \"%s\"\r\n" "$(VERSION_VER)" > version.h
+	@printf "#define TMD_CREATOR_DATE \"%s\"\r\n" "$(VERSION_DATE)" >> version.h
 
 %.o: %.cpp version.h
 	$(CXX) $(CFLAGS) -c $< -o $@

--- a/Makefile
+++ b/Makefile
@@ -10,6 +10,11 @@ ifeq ($(OS),Windows_NT)
 	TARGET = $(TARGET_NAME).exe
 	LFLAGS += -static -static-libgcc
 else
+	UNAME_S := $(shell uname -s)
+	ifeq ($(UNAME_S),Darwin)
+		CFLAGS += -I/usr/local/opt/openssl/include
+		LFLAGS += -L/usr/local/opt/openssl/lib
+	endif
 	TARGET = $(TARGET_NAME)
 endif
 

--- a/maketmd.cpp
+++ b/maketmd.cpp
@@ -41,6 +41,11 @@ distribution.
 uint32_t __bswap_32(uint32_t value) {
 	return _byteswap_ulong(value);
 }
+#elif __APPLE__
+#include <libkern/OSByteOrder.h>
+uint32_t __bswap_32(uint32_t value) {
+	return OSSwapInt32(value);
+}
 #else
 #include <byteswap.h>
 #endif	// _WIN32

